### PR TITLE
chore: migrate AWS SDK for JavaScript v2 APIs to v3

### PR DESCRIPTION
**Pull request by trivikr**
#### Summary

This PR migrates AWS SDK for JavaScript v2 APIs to v3 using [aws-sdk-js-codemod](https://www.npmjs.com/package/aws-sdk-js-codemod).

```console
$ npx aws-sdk-js-codemod@0.26.1 -t v2-to-v3 e2e/utils/artifacts.js
```

#### Ticket Link

From AWS SDK for JavaScript v2 [README](https://github.com/aws/aws-sdk-js):
> We are formalizing our plans to make the Maintenance Announcement (Phase 2) for AWS SDK for JavaScript v2 in 2023.

PR similar to https://github.com/mattermost/mattermost/pull/25301

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
- [ ] completed [Mattermost Contributor Agreement](https://mattermost.com/contribute/)
- [ ] executed `npm run lint:js` for proper code formatting

#### Device Information

N/A

#### Screenshots

N/A

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by trivikr** [comment]
cc active contributor @devinbinnie

**Review by saturninoabril — APPROVED** [review]
Thanks @trivikr, LGTM.

File: e2e/utils/artifacts.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const fs = require('fs');
 const path = require('path');
 
 const async = require('async');
-const AWS = require('aws-sdk');
+const { Upload } = require('@aws-sdk/lib-storage');
```
Comment (devinbinnie) [inline e2e/utils/artifacts.js:11]:
```suggestion
const {Upload} = require('@aws-sdk/lib-storage');
```

File: e2e/utils/artifacts.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ const fs = require('fs');
...
+const { S3 } = require('@aws-sdk/client-s3');
```
Comment (devinbinnie) [inline e2e/utils/artifacts.js:12]:
```suggestion
const {S3} = require('@aws-sdk/client-s3');
```

**Review by devinbinnie — APPROVED** [review]
LGTM! Just two small nitpicks (not sure why the linter didn't catch that)
Thanks @trivikr!

**Pull request merged**